lib/intercom/index.js

```diff
@@ -162,4 +164,38 @@ Intercom.prototype.track = function(track){
 
 function formatDate(date) {
   return Math.floor(date / 1000);
-}
+};
+
+/**
+ * Formats a company for use with intercom
+ *
+ * http://docs.intercom.io/#Companies
+ *
+ * TODO: add .companies()
+ *
+ * @param {Object} company
+ * @return {Object}
+ * @api private
+ */
+
+function formatCompany(company){
```

Comment:
Let's chat out rationale, but I think something similar to this approach will be less error-prone in the long run:


```js
var specialKeys = { name: 'name', id: 'id', created_at: 'created_at' };
function formatCompany(company){
  if (is.string(company)) {
    return [{ name: company }];
  }

  var results = reduce(company, function(acc, val, key) {
    var newKey = find(specialKeys, key);

    if (newKey) {
      acc[newKey] = val;
    } else {
      if (acc.custom_attributes) {
        acc.custom_attributes = { key: val };
      } else {
        acc.custom_attributes[key] = val;
      }
    }

    return results;
  }, {});

  return [results];
};
```